Bump zwave-js to 12.0.0 and zwave-js-server to 1.32.0 (#3227)
home-assistant/addons: zwave_js/CHANGELOG.md

# Changelog

## 0.1.91

> ### Breaking changes

Reviewer: I take it that the "heal" -> "rebuild routes" change haven't found their way into HA yet?

Author: it'll be handled transparently in 2023.10. It's not worth mentioning here because it doesn't impact device functionality

> - Firmware updates may no longer work properly in Home Assistant versions 2023.9.x and earlier due to a breaking change upstream that couldn't be made backwards compatible. They will be fixed in 2023.10.0 .

Reviewer: ```suggestion
- Firmware updates may no longer work properly in Home Assistant versions 2023.9.x and earlier due to a breaking change upstream that couldn't be made backwards compatible. They will be fixed in 2023.10.0.
```

Reviewer: I guess we should explain the reason:
Checking for updates no longer requires communication with the device and therefore reduces the risk of corrupting manufacturer information. This also means that updates for battery-powered devices can be detected without waking up the devices.

### Features

> - Detect an unresponsive stick and reset it

Reviewer: ```suggestion
- Unresponsive controllers are now detected and automatically restarted if possible
```

> - The default time after which battery-powered devices with no pending commands are sent back to sleep is now 250 ms (down from 1000ms)

Reviewer: ```suggestion
- Battery-powered devices are sent back to sleep after 250 ms with no command (down from 1000 ms). This should result in significant battery savings for devices that frequently wake up.
```

> - We no long rebuild the @serialport/bindings-cpp package from source

Reviewer: ```suggestion
- We no longer rebuild the @serialport/bindings-cpp package from source
```

> - Firmware updates may no longer work properly in Home Assistant versions 2023.9.x and earlier due to a breaking change upstream that couldn't be made backwards compatible. Firmware updates will work once again starting in 2023.10.0. As a result of this breaking change, Checking for updates no longer requires communication with the device and therefore reduces the risk of corrupting manufacturer information. This also means that updates for battery-powered devices can be detected without waking up the devices. 

Reviewer 2: ```suggestion
- Firmware updates may no longer work properly in Home Assistant versions 2023.9.x and earlier due to a breaking change upstream that couldn't be made backward compatible. Firmware updates will work once again starting in 2023.10.0. As a result of this breaking change, Checking for updates no longer requires communication with the device and, therefore, reduces the risk of corrupting manufacturer information. This also means that updates for battery-powered devices can be detected without waking up the devices.

```

> - Unresponsive controllers are now detected and automatically restarted if possible
> - Battery-powered devices are sent back to sleep after 250 ms with no command (down from 1000 ms). This should result in significant battery savings for devices that frequently wake up.
> - We no longer rebuild the @serialport/bindings-cpp package from source

Reviewer 2: ```suggestion
- Unresponsive controllers are now detected and automatically restarted if possible.
- Battery-powered devices are sent back to sleep after 250 ms with no command (down from 1000 ms). This should result in significant battery savings for devices that frequently wake up.
- We no longer rebuild the @serialport/bindings-cpp package from source.

```

### Bug fixes

- A bug in the 7.19.x SDK has surfaced where the controller gets stuck in the middle of a transmission. Previously this would go unnoticed because the failed commands would cause the nodes to be marked dead until the controller finally recovered. Since v11.12.0 however, Z-Wave JS would consider the controller jammed and retry the last command indefinitely. This situation is now detected and Z-Wave JS attempts to recover by soft-resetting the controller when this happens.

> - Removed auto-disabling of soft-reset capability

Reviewer: ```suggestion
- Fixed an issue where supporting controllers would no longer be automatically restarted after failing to do so once
```

> - Default to RF protection state Unprotected if not given for Protection CC V2+

Reviewer: I don't think this one is really worth mentioning.

> 

Reviewer: ```suggestion
- Devices that send notifications from endpoints, e.g. Aeotec Wallmote, are now properly supported
```
(at least on the driver side, what about HA?)

Author: this will be supported in 2023.10 although thanks for reminding me, I missed that

> - A bug in the 7.19.x SDK has surfaced where the controller gets stuck in the middle of a transmission. Previously this would go unnoticed because the failed commands would cause the nodes to be marked dead until the controller finally recovered. Since v11.12.0 however, Z-Wave JS would consider the controller jammed and retry the last command indefinitely. This situation is now detected and Z-Wave JS attempts to recover by soft-resetting the controller when this happens.
> - Fixed an issue where supporting controllers would no longer be automatically restarted after failing to do so once
> - Devices that send notifications from endpoints, e.g. Aeotec Wallmote, are now properly supported

Reviewer 2: ```suggestion
- A bug in the 7.19.x SDK has surfaced where the controller gets stuck in the middle of a transmission. Previously, this would go unnoticed because the failed commands would cause the nodes to be marked dead until the controller finally recovered. Since v11.12.0, however, Z-Wave JS would consider the controller jammed and retry the last command indefinitely. This situation is now detected, and Z-Wave JS attempts to recover by soft resetting the controller when this happens.
- Fixed an issue where supporting controllers would no longer be automatically restarted after failing to do so once.
- Devices that send notifications from endpoints, like Aeotec Wallmote, are now properly supported.

```
